qode/integration/node_integration.cc

```diff
@@ -113,6 +124,9 @@ void NodeIntegration::EmbedThreadRunner(void *arg) {
 
     // Deal with event in main thread.
     self->WakeupMainThread();
+
+    // Wait for the main loop to deal with events.
+    uv_sem_wait(&self->embed_sem_);
   }
```

Comment:
If we skip the batch processing and only do this change where we wait for main loop to deal with events after main thread has woken up does it achive similar results?
If not lets go with batch processing.

Reply:
There are two mostly separate things in this PR:

* The moving of the semaphore is the most important part here as it fixes Windows.
* Processing multiple UV events each time (batch).

The batch is optional, but the impact on performance is quite great if you are doing I/O. I remember someone complaining in an issue somewhere that NodeGui's I/O is much much slower than plain nodejs.

(Yes, I probably should've made 2 PRs. my bad)